scripts/Calendar/EventSources/FreeFormEventsSource.ts

```diff
@@ -80,17 +102,31 @@ export class FreeFormEventsSource implements Calendar_Contracts.IEventSource {
     }
 
     public getCategories(query?: Calendar_Contracts.IEventQuery): IPromise<Calendar_Contracts.IEventCategory[]> {
-        return this.getEvents().then(
-            (events: Calendar_Contracts.CalendarEvent[]) => {
-                return this._categorizeEvents(events, query);
-            });
+        var deferred = Q.defer();
+        VSS.getService("ms.vss-web.data-service").then((extensionDataService: Services_ExtensionData.ExtensionDataService) => {
+           extensionDataService.queryCollectionNames([this._teamId]).then((collections: Contributions_Contracts.ExtensionDataCollection[]) => {
```

Comment:
I would store the categories in a separate document collection from the events. Maybe the id for events is the team id, but for categories it is "{teamId}-categories" ?